vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -617,11 +617,12 @@ - (void)onGrantFinish:(ledger::Result)result grant:(ledger::GrantPtr)grant
   return bridgedReports;
 }
 
-- (BATBalanceReportInfo *)balanceReportForMonth:(BATActivityMonth)month year:(int)year
+- (void)balanceReportForMonth:(BATActivityMonth)month year:(int)year completion:(void (NS_NOESCAPE ^)(bool result, BATBalanceReportInfo * _Nullable info))completion
```

Comment:
Let's just drop the `bool result` in the completion block. We'll know whether or not it failed based on if `info` is nil or not.

Reply:
As it is, info will never be nil per `bat_publishers.cc`: 
```
void BatPublishers::getBalanceReport(
    ledger::ACTIVITY_MONTH month,
    int year,
    ledger::GetBalanceReportCallback callback) {
  ledger::BalanceReportInfo info;
  bool result =
    getBalanceReportInternal(month, year, &info);
  callback(result, info.Clone());
}
```

Are you suggesting it should be: 
```
void BatPublishers::getBalanceReport(
    ledger::ACTIVITY_MONTH month,
    int year,
    ledger::GetBalanceReportCallback callback) {
  ledger::BalanceReportInfo info;
  bool result =
    getBalanceReportInternal(month, year, &info);
  callback(result ? info.Clone() : nullptr);
}
```
?

Reply:
I didn't necessarily mean the underlying C++ had to change, just that for in Obj-C we use nullability checks a lot, so its simpler. (In Obj-C calling or accessing `nil` doesn't result in a crash).

So just meant that for the Obj-C completion block can just be called like `completion(result ? bridgedInfo : nil)`

Reply:
gotcha, fixed